controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -155,12 +156,20 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 	// If the machine that is about to be deleted is the etcd leader, move it to the newest member available.
 	if controlPlane.IsEtcdManaged() {
 		etcdLeaderCandidate := controlPlane.HealthyMachines().Newest()
+		if etcdLeaderCandidate == nil {
+			err := errors.New("no healthy etcd leadership candidate machines found")
+			log.Error(err, "Failed to move etcd leadership")
+			conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, err.Error())
+			return ctrl.Result{}, err
```

Comment:
should this be
```suggestion
			err := errors.New("no healthy etcd leadership candidate machines found")
			log.Info("A control plane machine needs remediation, but there is no healthy machine to forward etcd leadership to")
			conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, "KCP can't remediate this machine because there is no healthy machine to forward etcd leadership to")
			return ctrl.Result{}, nil
```
So we handle the condition, surface it but we do not go into error loopback, like all the other conditions where we cannot remediate?